### ISSUE TITLE
Not honoring Version - always installs 3.6.8

### ISSUE THREAD

**Issue by wjdavis5** [rabbitmq/chef-cookbook#480]
Ubuntu 16.04 / vagrant / chef solo

```
node.default['rabbitmq']['use_distro_version'] = false
node.default['rabbitmq']['version'] = '3.7.3'

#other settings

include_recipe 'rabbitmq::default'
include_recipe 'rabbitmq::mgmt_console'
```

This always seems to install 3.6.8:

```
vagrant@ubuntu1604:~$ sudo rabbitmqctl status
Status of node rabbit@ubuntu1604 ...
[{pid,18419},
 {running_applications,
     [{rabbitmq_management_visualiser,"RabbitMQ Visualiser","3.6.8"},
      {rabbitmq_management,"RabbitMQ Management Console","3.6.8"},
      {rabbitmq_web_dispatch,"RabbitMQ Web Dispatcher","3.6.8"},
      {rabbitmq_management_agent,"RabbitMQ Management Agent","3.6.8"},
      {rabbit,"RabbitMQ","3.6.8"},
      {cowboy,"Small, fast, modular HTTP server.","1.0.4"},
      {ranch,"Socket acceptor pool for TCP protocols.","1.3.0"},
      {os_mon,"CPO  CXC 138 46","2.4"},
      {amqp_client,"RabbitMQ AMQP Client","3.6.8"},
      {ssl,"Erlang/OTP SSL application","7.3"},
      {rabbit_common,
          "Modules shared by rabbitmq-server and rabbitmq-erlang-client",
          "3.6.8"},
      {syntax_tools,"Syntax tools","1.7"},
      {public_key,"Public key infrastructure","1.1.1"},
      {cowlib,"Support library for manipulating Web protocols.","1.0.2"},
      {crypto,"CRYPTO","3.6.3"},
      {asn1,"The Erlang ASN1 compiler version 4.0.2","4.0.2"},
      {compiler,"ERTS  CXC 138 10","6.0.3"},
      {mnesia,"MNESIA  CXC 138 12","4.13.3"},
      {xmerl,"XML parser","1.3.10"},
      {inets,"INETS  CXC 138 49","6.2"},
      {sasl,"SASL  CXC 138 11","2.7"},
      {stdlib,"ERTS  CXC 138 10","2.8"},
      {kernel,"ERTS  CXC 138 10","4.2"}]},
 {os,{unix,linux}},
 {erlang_version,
     "Erlang/OTP 18 [erts-7.3] [source] [64-bit] [smp:2:2] [async-threads:64] [kernel-poll:true]\n"},
 {memory,
     [{total,59439072},
      {connection_readers,0},
      {connection_writers,0},
      {connection_channels,0},
      {connection_other,2712},
      {queue_procs,2712},
      {queue_slave_procs,0},
      {plugins,589464},
      {other_proc,21756048},
      {mnesia,76576},
      {metrics,194088},
      {mgmt_db,149832},
      {msg_index,77056},
      {other_ets,2510000},
      {binary,117608},
      {code,27537508},
      {atom,992409},
      {other_system,5624435}]},
 {alarms,[]},
 {listeners,
     [{clustering,25672,"::"},
      {amqp,5672,"::"},
      {'amqp/ssl',5671,"::"},
      {https,15671,"::"}]},
 {vm_memory_high_watermark,0.81},
 {vm_memory_limit,1684191191},
 {disk_free_limit,50000000},
 {disk_free,24790769664},
 {file_descriptors,
     [{total_limit,924},{total_used,2},{sockets_limit,829},{sockets_used,0}]},
 {processes,[{limit,1048576},{used,329}]},
 {run_queue,0},
 {uptime,26},
 {kernel,{net_ticktime,60}}]
vagrant@ubuntu1604:~$
```

**Comment by michaelklishin** [comment]
We cannot help with the amount of information provided. The version attribute [is used](https://github.com/rabbitmq/chef-cookbook/blob/master/recipes/default.rb#L32).

Use Chef client debug logging to find out more.

**Comment by michaelklishin** [comment]
3.6.8 is [not mentioned anywhere in the code](https://github.com/rabbitmq/chef-cookbook/search?utf8=%E2%9C%93&q=3.6.8&type=) and is not a [default version in Ubuntu 16.04](https://packages.ubuntu.com/search?keywords=rabbitmq-server&searchon=names&suite=all&section=all) (or any other version).

Very likely this is something environment-specific.

**Comment by michaelklishin** [comment]
3.6.8 was provisioned by default [in 5.2.x](https://github.com/rabbitmq/chef-cookbook/commit/e91e9e3c2f6b367101ccf41e0c741c11ffa212ef). There were 3 releases since then + changes in master. Please give 5.4.1 a try.

**Comment by michaelklishin** [comment]
Another thing I'm not sure about is why you override `node.default` attributes instead of `node`/`node.normal`.

**Comment by wjdavis5** [comment]
Thanks for the info - when I updated my cookbooks I do see that it updated to 3.6.14.
And also it is now honoring the version.
Regarding node / node.nor111mal - just me being a chef noob.

This was my fault, I thought our build was calling `berks update`  so I just assumed I was working off the latest version.

However during this I discovered it is in fact failing to download and install 3.7.3 - I'll open a new issue to track that.
